numpy/distutils/misc_util.py

```diff
@@ -2287,6 +2299,16 @@ def generate_config_py(target):
     f.write('# This file is generated by %s\n' % (os.path.abspath(sys.argv[0])))
     f.write('# It contains system_info results at the time of building this package.\n')
     f.write('__all__ = ["get_info","show"]\n\n')
+
+    if system_info.shared_libs:
+        f.write("""
+
+import os
+
+os.environ["PATH"] += os.pathsep + os.path.join(os.path.dirname(__file__), '_lib')
```

Comment:
This is only added when DLLs are generated and the generation code places the DLLs in the _lib directory and creates the lib directory if it does not exist.

All of this only happens if there is fortran code and mingw is available and ifortran is not available and the compiler is msvc. In other words, if there is not other option but to create the lib directory and place the DLLs in it.